src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -1,33 +1,43 @@
 <script lang="ts">
 	import { addressStore } from '$lib/stores/address.store';
-	import { isNullish } from '@dfinity/utils';
-	import { certifyEthAddress } from '$lib/services/address.services';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { certifyBtcAddressMainnet, certifyEthAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
-	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
+	import {
+		addressesCertified,
+		btcAddressMainnet,
+		btcAddressMainnetNotCertified,
+		ethAddress,
+		ethAddressNotCertified
+	} from '$lib/derived/address.derived';
+	import { NETWORK_BITCOIN_ENABLED } from '$env/networks.btc.env.js';
 
 	const validateAddress = async () => {
-		const ethAddressData = $addressStore?.[ETHEREUM_TOKEN_ID];
-
-		if (isNullish(ethAddressData)) {
-			// No address is loaded, we don't have to verify it
```

Comment:
This check was removed?

Reply:
i included it in the rest of the code, but inverted `if (nonnullish(....`

Reply:
actually, let me leave it there